gui/src/2D/controls/sliders/baseSlider.ts

```diff
@@ -301,7 +301,7 @@ export class BaseSlider extends Control {
             value = this._minimum + ((x - this._currentMeasure.left) / this._currentMeasure.width) * (this._maximum - this._minimum);
         }
 
-        const mult = (1 / this._step) | 0;
+        const mult = (1.0 / this._step);
         this.value = this._step ? ((value * mult) | 0) / mult : value;

```

Comment:
I don't understand this formula (I know it is not changed by this PR)...

Either we want `this.value` to be independent from `this._step` and then `this.value = value` or we want `this.value` to be a multiple of `this._step`, in which case the formula is wrong and should be something like:

```javascript
this.value = this._step ? ((value / this._step) | 0) * this._step : value;
```
I think we are in the latter case when `this._step != 0` because we are in the former case when `this._step == 0`.

Reply:
I tend to agree

Reply:
Yea I agree this makes a lot more sense.